doc/reference/plutus/reference-script-example.md

```diff
@@ -0,0 +1,146 @@
+
+A reference script is a script that exists at a particular transaction output. It can be used to witness, for example, a UTxO at the corresponding script address of said reference script. Why is this useful? It means we no longer have to include the script in the transaction, greatly reducing its size
+
+### An example of using a Plutus V2 reference script
+
+Below is an example that shows how to use a Plutus spending script. We will walk though the [shell script example of how to use a reference script to spend a tx input](scripts/plutus/example-reference-script-usage.sh).This is a step-by-step process involving:
+
++ the creation of the `Required Redeemer` Plutus txin script
++ the creation of the `Required Redeemer` Plutus script at a transaction output (creation of the reference script)
++ sending ADA and a datum to the Plutus script address
++ spending ADA at the Plutus script address using the Plutus reference script
+
+In this example we will use the [Required Redeemer](scripts/plutus/scripts/v2/required-redeemer.plutus) Plutus spending script. In order to execute a reference Plutus spending script, we require the following:
+
+- Collateral tx input(s) - these are provided and are forfeited in the event the Plutus script fails to execute.
+- A Plutus tx output with accompanying datum hash. This is the tx output that sits at the Plutus script address. It must have a datum hash, otherwise, it is unspendable.
+- The reference transaction input containing the corresponding Plutus script. We must create the transaction output containing the reference Plutus script.
+
+#### Creating the `Required Redeemer` Plutus spending script
+
+The plutus-example executable will automagically generate several Plutus scripts in the CLI-compatible text envelope format.
+
+Run the following commands:
+
+```bash
+git clone git@github.com:input-output-hk/plutus-apps.git
+
+cd plutus-apps/plutus-example
+
+cabal run exe:plutus-example
+```
+
+This will output `required-redeemer.plutus` in the `generated-plutus-scripts/v2` dir.
+
+#### Setting up a local Babbage node cluster
+
+There is a convenient script that will set up a Babbage cluster immediately on your local machine.
+
+Run the following command:
+
+```bash
+cabal install cardano-cli
+cabal install cardano-node
+./scripts/babbage/mkfiles.sh
+```
+
+To start your babbage cluster you need to run the `example/run/all.sh` shell script.
+The remainder of this guide briefly talks about the [shell script example](scripts/plutus/example-reference-script-usage.sh) that automatically does everything for you.
+
+#### Creating a reference script at a transaction output and
+#### sending ADA to the script address (with a datum)
+
+In order to use a reference script we must first create said reference script at a particular address. Note you cannot create a reference script at the same correspnding script address, this will result in a failure when trying to spend the transaction input at the script address.
```

Comment:
'In order to use a reference script, we must first create this script at a particular address. Note that you cannot create a reference script at the same corresponding script address as this will result in a failure when trying to spend the transaction input at the script address.'